code/game/turfs/space/space.dm

```diff
@@ -129,11 +129,9 @@
 	if ((!(A) || src != A.loc))
 		return
 
-	if(destination_z)
+	if(destination_z && destination_x && destination_y)
 		var/old_z = A.z
-		A.x = destination_x
-		A.y = destination_y
-		A.z = destination_z
+		A.forceMove(locate(destination_x, destination_y, destination_z))
 		if (old_z != destination_z)
```

Comment:
If you're using forceMove, remove the onTransitZ call, as it is handled by forceMove